backend/src/test/java/ca/gov/dtsstn/cdcp/api/service/UserServiceTests.java

```diff
@@ -85,4 +88,34 @@ void testCreateConfirmationCodeForUser() {
 		assertThat(confirmationCode.getExpiryDate()).isAfter(Instant.now());
 	}
 
+	@Test
+	@DisplayName("Test userService.verifyConfirmationCode(..) with valid input")
+	void testVerifyConfirmationCode_Valid() {
+		final var creationDate = LocalDateTime.now().minusDays(73).toInstant(ZoneOffset.UTC);
+		final var expiryDate = LocalDateTime.now().plusDays(288).toInstant(ZoneOffset.UTC);
+		final var codeValue = "code value";
+		final var confirmationCode = ImmutableConfirmationCode.builder().code(codeValue).createdDate(creationDate).expiryDate(expiryDate).build();
+		final var user = ImmutableUser.builder().addConfirmationCodes(confirmationCode).build();
+
+		assertThat(userService.verifyConfirmationCode(codeValue, user)).isTrue();
```

Comment:
As mentioned in other comments.. don't pass a user object into this method. Pass a user id and make the method look that user up in the database.

Reply:
Implemented in commit 2cebd2d6f0685660fa512c66c4a5da16869070e1.